src/hotspot/share/oops/constantPool.cpp

```diff
@@ -138,7 +138,11 @@ void ConstantPool::metaspace_pointers_do(MetaspaceClosure* it) {
   log_trace(cds)("Iter(ConstantPool): %p", this);
 
   it->push(&_tags, MetaspaceClosure::_writable);
-  it->push(&_cache);
+  if (!pool_holder()->is_rewritten()) {
+    it->push(&_cache, MetaspaceClosure::_writable);
```

Comment:
This shouldn't be necessary. All ConstantPoolCache are allocated in RW space by default.

Reply:
Fixed.